dot/system/service.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2020 ChainSafe Systems (ON) Corp.
+// This file is part of gossamer.
+//
+// The gossamer library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The gossamer library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+
+package system
+
+import "github.com/ChainSafe/gossamer/dot/types"
+
+// Service struct to hold rpc service data
+type Service struct {
```

Comment:
this looks good! however, it doesn't look like it's started when the node is started in dot `NewNode`?

Reply:
I don't have this started because there is no go routine to start, so I guess it's not really a service.  It's currently a place to hold static chain and app values.  Should I implement the Service interface to add the Start and Stop functions and add call to start so it's treated like other services (for consistency, and in case we add a go routine later).  OR should I treat this differently since it's not really a service, and if so, how?

Reply:
Should it be part of core?

Reply:
maybe just rename it from `Service` to `Info` or something like that, since there isn't a need to make it into an actual service

Reply:
Hmmm... Not entirely sure at the moment. If you think it is going to be a service (adhere to the service interface) and should be its own service, I think it works as is.

Reply:
I made it implement the Service interface, so it's handled like the others.  I was running into issues with cyclic references when I tried making it a part of rpc service.  This seems ok since it's following the pattern (now) that we're using for the others.